utility/check_baseline.py

```diff
@@ -0,0 +1,202 @@
+    ants[54]=katpoint.Antenna('m054, -30:42:39.8, 21:26:38.0, 1086.6, 13.5,  871.980  -499.812  13.364')
+    ants[55]=katpoint.Antenna('m055, -30:42:39.8, 21:26:38.0, 1086.6, 13.5,  1201.780 96.492  10.023')
+    ants[56]=katpoint.Antenna('m056, -30:42:39.8, 21:26:38.0, 1086.6, 13.5,  1598.403  466.668 6.990')
+    ants[57]=katpoint.Antenna('m057, -30:42:39.8, 21:26:38.0, 1086.6, 13.5,  294.645  3259.915  -10.637')
+    ants[58]=katpoint.Antenna('m058, -30:42:39.8, 21:26:38.0, 1086.6, 13.5,  2805.764  2686.873  -3.660')
+    ants[59]=katpoint.Antenna('m059, -30:42:39.8, 21:26:38.0, 1086.6, 13.5,  3686.427  758.895 11.822')
+    #
+    ants[60]=katpoint.Antenna('m060, -30:42:39.8, 21:26:38.0, 1086.6, 13.5,  3419.683 -1840.478 23.697')
+    ants[61]=katpoint.Antenna('m061, -30:42:39.8, 21:26:38.0, 1086.6, 13.5,  -16.409 -2323.779 21.304')
+    ants[62]=katpoint.Antenna('m062, -30:42:39.8, 21:26:38.0, 1086.6, 13.5,  -1440.632 -2503.773  21.683')
+    ants[63]=katpoint.Antenna('m063, -30:42:39.8, 21:26:38.0, 1086.6, 13.5,  -3419.585 -1840.480 16.383')
+    return ants
+
+
+def get_uvw(target,ants,timevalues):
+    # Set up empty uvw then loop through the half the antennas (n,m)
+    # and calculate u,v,w for the target and antennas n and m
+    u,v,w=[],[],[]
+    for n in range(np.shape(ants)[0]-1):
+        for m in range(n+1,np.shape(ants)[0]):
+            u_new,v_new,w_new=target.uvw(ants[n], timevalues ,ants[m])
+            u.extend(u_new)
+            v.extend(v_new)
+            w.extend(w_new)
+    return np.array(u),np.array(v),np.array(w)
+    #u,v,w = get_uvw(target, ants, katpoint.Timestamp(str(time1)).secs + np.linspace(0.0, upt2 * 60. * 60., 360))
+
+def get_xyz(ants):
+    # Set up empty xyz then loop through the half the antennas (n,m)   
+    baselineno = (np.shape(ants)[0]* (np.shape(ants)[0]-1))//2
+    x,y,z=np.zeros((baselineno)),np.zeros((baselineno)),np.zeros((baselineno))
+    i=0
+    for n in range(np.shape(ants)[0]-1):
+        for m in range(n+1,np.shape(ants)[0]):
+            x[i],y[i],z[i]= np.array(ants[n].position_enu)-np.array(ants[m].position_enu)
+            i = i+ 1
+    return x,y,z
+
+def verbosef(string,verbosebool):
+    if verbosebool :
+        print(string)
+        
+def test_array(ants,reduced_ants,minfrac=None,mintotal=None,verbose=False,bins =None ):
+    observation_good = True
+    x,y,z = get_xyz(ants)
+    x1,y1,z1 = get_xyz(reduced_ants)
+    verbosef("There are %i antennas."%(len(reduced_ants)),verbose)
+    if mintotal is not None:
+        if len(reduced_ants) < mintotal:
+            observation_good = False
+            verbosef("There are not enough antennas. Only %i antennas avalible."%(len(reduced_ants)),verbose)
+            verbosef("No fewer than %i antennas."%(mintotal),verbose)
+    else:
+        verbosef("NO constraints on the number of antennas",verbose)
+
+    if bins is not None:
+        total = np.histogram((np.sqrt(x**2+y**2)),bins=bins)[0]
```

Comment:
Why do you ignore the difference in Up - not important because the array is so flat?

Reply:
That is a good point. But if I were doing this properly I would use projected baselines.  It is mainly due to the PI's only specifying the criteria in terms of the 2D coordinates .